buildSrc/src/main/kotlin/buildsrc/convention/maven-publishing.gradle.kts

```diff
@@ -0,0 +1,153 @@
+package buildsrc.convention
```

Comment:
`maven-publishing.gradle.kts` is a bit more complicated than the other convention plugins, because it involves conditionally configuring publishing and setting up the `signing` pluging.

I've found the `signing` plugin doesn't play very well with Gradle task-config avoidance, so we'll have to check carefully that it still works.

Worst comes to worst the publishing and signing config can be moved back to the root `build.gradle.kts`

Reply:
I gave up on trying to make this plugin applicable to both JVM and Multiplatform, and instead made a `publish-jvm` convention.